src/server.cpp

```diff
@@ -1,17 +1,79 @@
 #include "crow.h"
+#include <string>
+#include <vector>
+#include <boost/thread.hpp>
+
+std::vector<std::string> locks;
+boost::mutex searchLock;
+
+bool getLock(std::string lockName) 
+{
+  searchLock.lock();
+  for (long unsigned int i = 0; i < locks.size(); i++) 
```

Comment:
Tip: the array of locks _always_ contains unique values (strings). In that case, you should always use a [set](https://www.cplusplus.com/reference/set/set/). 
Looking up, deleting, and inserting elements is easier (and faster!) when you use sets. We can go over the reasons for using a set later!

Note: a [map](https://www.cplusplus.com/reference/map/map/) is similar to a set.